wandb/sdk/wandb_init.py

```diff
@@ -859,23 +859,36 @@ def init(  # noqa: C901
             f"\nconfig: {config.base_no_artifacts}"
         )
 
-        if wandb.run is not None and os.getpid() == wandb.run._init_pid:
+        if previous_run := self._wl.most_recent_active_run:
             if (
                 settings.reinit in (True, "finish_previous")
                 # calling wandb.init() in notebooks finishes previous runs
                 # by default for user convenience.
                 or (settings.reinit == "default" and wb_ipython.in_notebook())
             ):
-                self._logger.info("finishing previous run: %s", wandb.run.id)
-                wandb.run.finish()
+                run_printer.display(
+                    "Finishing previous runs because reinit is set"
```

Comment:
i think it is useful to print the run id, or some other identifier of the run, given that there might be several

Reply:
Each run will print a footer as it completes which has that info; albeit if `run.finish()` is stuck, the user won't know which run it corresponds to, but maybe we should update `run.finish()` to include that info?

Reply:
yeah... just thinking how to make sure the message is as informative as possible to the user

Reply:
Can do that in a separate PR!